### PR TITLE
feat: include row count in SHOW TOPICS EXTENDED

### PR DESCRIPTION
**Pull request by big-andy-coates**
### Description 

While investigating issues I've always thought it would be useful if the output of `SHOW TOPICS EXTENDED` included the number of records within the topic.

With this change, it now does... ish!

The record count is calculated as the difference between the earliest and latest offsets. For non-compacted topics this shows the number of records in the topic.
For compacted topics this shows the maximum number of records that can be in the topic, though compaction may have reduced this number. This is why the number is reported as `Max` row count. It is intended as an indication only, i.e. to help with debugging etc.

### old output:

```
ksql> SHOW TOPICS EXTENDED;

 Kafka Topic                    | Partitions | Partition Replicas | Consumers | ConsumerGroups
----------------------------------------------------------------------------------------------
 default_ksql_processing_log    | 1          | 1                  | 0         | 0
 pageviews                      | 1          | 1                  | 0         | 0
 users                          | 1          | 1                  | 0         | 0
----------------------------------------------------------------------------------------------
```

### new output:

```
ksql> SHOW TOPICS EXTENDED;

 Kafka Topic                    | Partitions | Replicas | Max Row Count | Consumers | ConsumerGroups
----------------------------------------------------------------------------------------------------
 default_ksql_processing_log    | 1          | 1        | 0             | 0         | 0
 pageviews                      | 1          | 1        | 2345          | 0         | 0
 users                          | 1          | 1        | 25            | 0         | 0
-----------------------------------------------------------------------------------------------------
```

### Testing done 

Manual + unit + integration

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: docs/developer-guide/ksqldb-reference/show-topics.md [hunk 1]
```diff
@@ -30,31 +30,41 @@ and their active consumer counts.
 
 `SHOW ALL TOPICS` lists all topics, including hidden topics.
 
+`SHOW TOPICS EXTENDED` includes additional information about each topic, including the 
+`Max Row Count`. This is intended as a guide only. It is calculated as the sum of the difference 
+between each of topic partition's latest and earliest offset.
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/show-topics.md:35]:
```suggestion
between each of the topic partition's latest and earliest offset. 

```

File: docs/developer-guide/ksqldb-reference/show-topics.md [hunk 2]
```diff
@@ -30,31 +30,41 @@ and their active consumer counts.
 
 `SHOW ALL TOPICS` lists all topics, including hidden topics.
 
+`SHOW TOPICS EXTENDED` includes additional information about each topic, including the 
+`Max Row Count`. This is intended as a guide only. It is calculated as the sum of the difference 
+between each of topic partition's latest and earliest offset. 
+
+For non-compacted topics, for example those normally backing a stream, this number is the _actual_
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/show-topics.md:37]:
```suggestion
For non-compacted topics, like those normally backing a stream, this number is the _actual_ 

```

File: docs/developer-guide/ksqldb-reference/show-topics.md [hunk 3]
```diff
@@ -30,31 +30,41 @@ and their active consumer counts.
...
 `SHOW ALL TOPICS` lists all topics, including hidden topics.
 
+`SHOW TOPICS EXTENDED` includes additional information about each topic, including the 
+`Max Row Count`. This is intended as a guide only. It is calculated as the sum of the difference 
+between each of topic partition's latest and earliest offset. 
+
+For non-compacted topics, for example those normally backing a stream, this number is the _actual_ 
+row count in the topic. 
+
+For compacted topics, for example those normally backing a table, this number is the _maximum_ row
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/show-topics.md:40]:
```suggestion
For compacted topics, like those normally backing a table, this number is the _maximum_ row 

```

**Comment by agavra** [comment]
@big-andy-coates please look at #3604 - not sure if there is overlap or not, havent looked at this PR yet

**Comment by big-andy-coates** [comment]
> @big-andy-coates please look at #5476 - not sure if there is overlap or not, havent looked at this PR yet

Not related...

**Comment by big-andy-coates** [comment]
Note to self: would be good to include compaction / retention settings in the output!  If not compacted can change `MAX ROW COUNT` to just `ROW COUNT`.  Maybe have topic level summary.

**Review by spena — APPROVED** [review]
Looks good!

What do you think of adding a new column to show whether the topic is `COMPACT` or not? And change the `Max Row Count` to only `Row Count`? Users would immediately understand the changes when showing the topics and seeing different row counts on compacted topics, right?

**Pull request closed without merging**